### ISSUE TITLE
Changing name of terminal with split terminals in `Switch Active Terminal` changes wrong terminal name

### ISSUE THREAD

**Issue by gioragutt** [Microsoft/vscode#48671]
- VSCode Version:
```
Version 1.22.2
Commit 3aeede733d9a3098f7b4bdc1f66b63b0f48c1ef9
Date 2018-04-12T16:38:45.278Z
Shell 1.7.12
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
```
- OS Version: `Windows 10`

Steps to Reproduce:

1. Open terminal (AKA `term1`) and split it (with `Ctrl+\`)
2. Open another terminal (AKA `term2`)
3. `Ctrl+Shift+P Switch Active Terminal`
4. Click on the `Rename` button of `term1` and change it's name, this changes fine: `bash, bash` -> `client, bash`
5. Click on the `Rename` button of `term2` and change it's name, this changes the name of the second split terminal of `term1`: `client, bash` -> `client, server`

![Bug is bad](https://user-images.githubusercontent.com/13711224/39240966-0bf1d44a-488e-11e8-89d2-4b5aa9c2e870.gif)

6. This happens when opening more terminals or split terminals.
Basically, the `Rename` action changes the terminal at `index = line in the list of terminals`,
but the list of terminals contains all the split terminals, so `index = 2` in the switch list is the second terminal in the terminals list, which is the split terminal of `term1` in the example.

**Comment by gioragutt** [comment]
This also happens with the `Delete` button, same idea. Deletes the wrong terminal.

**Comment by gioragutt** [comment]
@Tyrair can I get a comment on this?

**Comment by Tyriar** [comment]
@gioragutt sorry, working through a big backlog of notifications. I can repro the bug, will see if I can get it fixed in 1.24 👍

**Comment by Tyriar** [comment]
@gioragutt I fixed it, this was missed when the split concept was added. It will now rename/kill the _last active_  terminal (not necessarily the first) within the selected "tab".

**Comment by gioragutt** [comment]
@Tyriar First of all, thanks for the reply and work, that's immensely appreciated.

The change is acceptable,  although, you could just assign an index to each terminal and have it displayed in the dropdown.

f.e: `tab 1` contains terminals `1`, `2` and `3`, and `tab 2` contains `1` and `2`.

The dropdown could consist of:
* `1.1 <name>`
* `1.2 <name>`
* `1.3 <name>`
* `2.1 <name>`
* `2.2 <name>`

That's just as easy to implement, and might give more control over the terminals.
I haven't seen the source code for terminals, but if you iterate over tabs and you can get the terminals, or you can link each terminal to the index of it's tab, then that's quite straightforward.

@oriash93 תוקן 👍
